src/main/webapp/assets/js/index.js

```diff
@@ -98,20 +101,39 @@ function filterCategory(category) {
   document.getElementById(`filter-btn-${category}`).className += " active";
 
   // filter for products
-  let products = document.getElementsByClassName("product-listing-card");
+  let fullProducts = document.getElementsByClassName("product-listing-card");
+  var productsToLoad = []
```

Comment:
Missing ; also. I'm kind of surprised js doesn't throw compile errors for this?

Reply:
Javascript has "optional" semicolons. it's weird.